pkg/build/api/types.go

```diff
@@ -111,6 +111,10 @@ type BuildSource struct {
 	// This allows to have buildable sources in directory other than root of
 	// repository.
 	ContextDir string `json:"contextDir,omitempty"`
+
+	// SourceSecretName is the name of a Secret that would be used for setting
+	// up the authentication for cloning private repository.
+	SourceSecretName string `json:"sourceSecretName,omitempty"`
```

Comment:
ObjectRef would make sense to me..that would enable you to use a secret from a different namespace also.